components/funder/src/handler/handle_friend.rs

```diff
@@ -121,11 +121,14 @@ pub fn try_reset_channel<B>(
 fn forward_request<B>(
     m_state: &mut MutableFunderState<B>,
     send_commands: &mut SendCommands,
-    request_send_funds: RequestSendFundsOp,
+    mut request_send_funds: RequestSendFundsOp,
 ) where
     B: Clone + PartialEq + Eq + CanonicalSerialize + Debug,
 {
-    let next_pk = request_send_funds.route.index_to_pk(0).unwrap();
+    let next_pk = request_send_funds.route.index_to_pk(0).unwrap().clone();
+
+    // Remove the next node from remaining route.
+    request_send_funds.route.public_keys.remove(0);
```

Comment:
~~I think you might be able to remove and get the removed element in one line (And without a clone!)~~

Reply:
I think you should only remove the first node only before forwarding the request. Removing it here is probably too early. For example, this turns the `request_send_funds.route.is_empty()` check to be incorrect. 

Example layout: `B -- C -- D -- E`.
If we send a request from B to E, E should get a request with an empty route.
We should first check the length of the route, and only then possibly decide to remove the first node. Note that the first node should not be removed in the case of a the destination node receiving the request (Because the route is empty).

Reply:
I'm not sure if I'm thinking the right way, but here it's removing the node as late as possible, just before forwarding it. Before, that deletion was done just before calling `forward_request`, but then that function panicked as it couldn't get `next_pk` (as it was already deleted).

Reply:
I just looked at the full file, now I understand what happened. Removing the first node was moved into `forward_request()`. Could you take it back to just before the call to `forward_request()`? I want it to be next to be next to the part where the fees are taken, inside `handle_request_send_funds()`.